src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,206 @@
+     *
+     * @param id  The document id
+     * @param rev The document revision
+     * @return {@link DesignDocument}
+     */
+    public DesignDocument get(String id, String rev) {
+        assertNotEmpty(id, "id");
+        assertNotEmpty(id, "rev");
+        final URI uri = buildUri(dbc.getDBUri()).path(id).query("rev", rev).build();
+        return client.get(uri, DesignDocument.class);
+    }
+
+    /**
+     * Removes a design document from the database.
+     *
+     * @param id  The document id
+     * @return {@link DesignDocument}
+     */
+    public Response remove(String id) {
+        assertNotEmpty(id, "id");
+        final URI uri = buildUri(dbUri).path(id).build();
+        Response find = client.couchDbClient.get(uri, Response.class);
+        return dbc.remove(find.getId(), find.getRev());
+    }
+
+    /**
+     * Removes a design document using the id and rev from the database.
+     *
+     * @param id  The document id
+     * @param rev The document revision
+     * @return {@link DesignDocument}
+     */
+    public Response remove(String id, String rev) {
+        assertNotEmpty(id, "id");
+        assertNotEmpty(id, "rev");
+        return dbc.remove(id, rev);
+
+    }
+
+    /**
+     * Removes a design document using DesignDocument object from the database.
+     *
+     * @param designDocument the design document object to be removed
+     * @return {@link DesignDocument}
+     */
+    public Response remove(DesignDocument designDocument) {
+        assertNotEmpty(designDocument, "DesignDocument");
+        return dbc.remove(designDocument);
+    }
+
+    /**
+     * Deserialize a directory of javascript design documents to a DesignDocument object.
+     *
+     * @param directory the directory containing javascript files
+     * @return {@link DesignDocument}
+     */
+    public static List<DesignDocument> fromDirectory(File directory) {
+        List<DesignDocument> designDocuments = new ArrayList<DesignDocument>();
+        if(directory.isDirectory()) {
+            Collection<File> files = FileUtils.listFiles(directory, null, true);
+            for(File designDocFile : files) {
+                designDocuments.add(fromFile(designDocFile));
+            }
+        } else {
+            designDocuments.add(fromFile(directory));
+        }
+        return designDocuments;
+    }
+
+    /**
+     * Deserialize a javascript design document to a DesignDocument object.
+     *
+     * @param file the design document javascript file
+     * @return {@link DesignDocument}
+     */
+    public static DesignDocument fromFile(File file) {
+        assertNotEmpty(file, "Design js file");
+        DesignDocument dd = new DesignDocument();
+        Gson gson = new Gson();
+        //Deserialize JS file contents into DesignDocument object
+        try {
+            String contents = readURL(file.toURI().toURL());
+            dd = gson.fromJson(contents, DesignDocument.class);

```

Comment:
Instead of reading into a `String`, can't we just pass a `FileReader` into Gson?
Then we could also throw the `FileNotFoundException` back to the caller so they could handle figuring out if they had the wrong path or no read access or whatever.